frontend-react/src/components/Table/Table.test.tsx

```diff
@@ -2,8 +2,61 @@ import { fireEvent, screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 
 import { renderWithRouter } from "../../utils/CustomRenderUtils";
+import { mockFilterManager } from "../../hooks/filters/mocks/MockFilterManager";
+import { SortSettingsActionType } from "../../hooks/filters/UseSortOrder";
 
 import { TestTable } from "./TestTable";
+import Table, { ColumnConfig, TableConfig, TableRow } from "./Table";
+
+/* Table generation tools */
+
+const getSetOfRows = (count: number) => {
+    const testRows: TableRow[] = [];
+    for (let i = 0; i < count; i++) {
+        testRows.push({
+            id: i,
+            item: `Item ${i}`,
+            linkableItem: `UUID-${i}`,
+            mappedItem: i,
+            transformedValue: i,
+            sortedColumn: i,
+            locallySortedColumn: i,
+        });
+    }
+    return testRows;
+};
+
+const makeConfigs = (sampleRow: TableRow): ColumnConfig[] => {
+    const sampleMapper = new Map<number, string>([[2, "Mapped Item"]]);
+    const transformFunc = (v: any) => {
+        return v === 9 ? "Transformed Value" : v;
+    };
+    return Object.keys(sampleRow).map((key) => {
+        return {
+            dataAttr: key,
+            columnHeader: `${key[0].toUpperCase()}${key
+                .slice(1)
+                .toLowerCase()}`,
+            link: key.includes("link"),
+            linkBasePath: key.includes("link") ? "/base/" : undefined,
+            valueMap: key.includes("map") ? sampleMapper : undefined,
+            transform: key.includes("transform") ? transformFunc : undefined,
+            sortable: key.includes("sort") || key.includes("Sort"),
+            localSort: key.includes("local"),
+        } as ColumnConfig;
+    });
+};
+
+const getTestConfig = (rowCount: number): TableConfig => {
```

Comment:
I'm hoping these helpers will let us ween off of `TestTable` so we can ditch it altogether.